klayout/drc/rule_decks/main.drc

```diff
@@ -14,880 +14,763 @@
+       else
+         'true'
+       end
 
-logger.info("Gold enabled  %s" % [GOLD])
+logger.info("Gold enabled: #{GOLD}")
 
-if $mim_option
-  MIM_OPTION = $mim_option
-else
-  MIM_OPTION = "B"
-end
+# MIM
+MIM_OPTION = $mim_option || 'B'
 
-logger.info("MIM Option selected %s" % [MIM_OPTION])
+logger.info("MIM Option selected: #{MIM_OPTION}")
 
 # OFFGRID
-if $offgrid == "false"
-  OFFGRID = false
-else
-  OFFGRID = true
-end # OFFGRID
+OFFGRID = $offgrid != 'false'
 
-logger.info("Offgrid enabled  %s" % [OFFGRID])
+logger.info("Offgrid enabled:  #{OFFGRID}")
 
-if $thr
-    threads($thr)
-else
-    thr ||=  Etc.nprocessors
-    threads(thr)
-end
-logger.info("Number of threads to use %s" % [$thr])
+# threads
+$thr ||= Etc.nprocessors
+threads($thr)
+
+logger.info("Number of threads to use #{$thr}")
 
 #=== PRINT DETAILS ===
 logger.info("Verbose mode: #{$verbose}")
-if $verbose == "true"
+if $verbose == 'true'
   verbose(true)
 else
   verbose(false)
 end
 
 # === TILING MODE ===
-if $run_mode == "tiling"
+case $run_mode
+when 'tiling'
   tiles(500.um)
   tile_borders(10.um)
-  logger.info("Tiling  mode is enabled.")
+  logger.info('Tiling  mode is enabled.')
 
-elsif $run_mode == "deep"
+when 'deep'
   #=== HIER MODE ===
   deep
-  logger.info("deep  mode is enabled.")
+  logger.info('deep  mode is enabled.')
 else
   #=== FLAT MODE ===
   flat
-  logger.info("flat  mode is enabled.")
-end # run_mode
+  logger.info('flat  mode is enabled.')
+end
 
 # METAL_TOP
-if $metal_top
-  METAL_TOP = $metal_top
-else
-  METAL_TOP = "9K"
-end # METAL_TOP
+METAL_TOP = $metal_top || '9K'
 
-logger.info("METAL_TOP Selected is %s" % [METAL_TOP])
+logger.info("METAL_TOP Selected is #{METAL_TOP}")
 
 # METAL_LEVEL
-if $metal_level
-  METAL_LEVEL = $metal_level
-else
-  METAL_LEVEL = "5LM"
-end # METAL_LEVEL
+METAL_LEVEL = $metal_level || '5LM'
 
-logger.info("METAL_STACK Selected is %s" % [METAL_LEVEL])
+logger.info("METAL_STACK Selected is #{METAL_LEVEL}")
 
 # FEOL
-if $feol == "false"
+if $feol == 'false'
   FEOL = $feol
-  logger.info("FEOL is disabled.")
+  logger.info('FEOL is disabled.')
 else
-  FEOL = "true"
-  logger.info("FEOL is enabled.")
-end # FEOL
+  FEOL = 'true'.freeze
+  logger.info('FEOL is enabled.')
+end
 
 # BEOL
-if $beol == "false"
+if $beol == 'false'
   BEOL = $beol
-  logger.info("BEOL is disabled.")
+  logger.info('BEOL is disabled.')
 else
-  BEOL = "true"
-  logger.info("BEOL is enabled.")
-end # BEOL
+  BEOL = 'true'.freeze
+  logger.info('BEOL is enabled.')
+end
 
 #================================================
 #------------- LAYERS DEFINITIONS ---------------
 #================================================
 polygons_count = 0
-logger.info("Read in polygons from layers.")
+logger.info('Read in polygons from layers.')
 
-def get_polygons(l, d)
-  ps = polygons(l, d)
-  return $run_mode == "deep" ? ps : ps.merged
+def get_polygons(layer, data_type)
+  ps = polygons(layer, data_type)
+  $run_mode == 'deep' ? ps : ps.merged
```

Comment:
shouldn't we keep the return here?

Reply:
No need, It works well without it.
![image](https://user-images.githubusercontent.com/91015308/217468535-567357ae-a135-4150-99c9-de9297ec8fdc.png)